### ISSUE TITLE
Where does the file with the suffix ".art, .fdm, .cir, .ecc" required in LoadFeature. M come from?

### ISSUE THREAD

**Issue by uptownfunkmars** [gaolinorange/Automatic-Unpaired-Shape-Deformation-Transfer#7]
Where does the file with the suffix ".art, .fdm, .cir, .ecc" required in LoadFeature. M come from?

**Comment by tommaoer** [comment]
Thanks for your attention to our work. These files are the shape descriptors of each mesh, which will be used in the Similarity Network for estimating the difference between two meshes. Then, you can obtain these files by the following steps:
1. use the script 'list.m' to generate a 'txt' file including the absolute path of objs, and copy it to the folder 'lightfield'.
2. run '3Dalignment.exe' to obtain the ".art .fdm .cir .ecc" files. You must push the key 'm' and then the OpenGL window will pop up for computing these files.
Please visit the [link](https://github.com/Sunwinds/ShapeDescriptor/tree/43d2338117c0f34c75e251e2472965ba7e37e55d/LightField/3DRetrieval_v1.8/3DRetrieval_v1.8) for more details.
If you have some questions, please feel free to contact me.

**Comment by uptownfunkmars** [comment]
@tommaoer Thanks ! i'll try it.

**Comment by sjz-suyi** [comment]
> @tommaoer Thanks ! i'll try it.

Did you succeed?sir.

**Comment by uptownfunkmars** [comment]
@sjz-suyi https://github.com/Sunwinds/ShapeDescriptor/tree/43d2338117c0f34c75e251e2472965ba7e37e55d/LightField/3DRetrieval_v1.8/3DRetrieval_v1.8/Executable
This link explains how to use it in detail.

**Comment by uptownfunkmars** [comment]
@tommaoer Is ACAPOpt.exe the same as the genfeature.exe in the [link](https://github.com/aldehydecho/convMesh/tree/master/genfeature)? And I run ACAPOpt.exe but didn't generate FeatureMat1.mat.

**Comment by tommaoer** [comment]
yes, it is the same as in the [link](https://github.com/aldehydecho/convMesh/tree/master/genfeature). It extracts the deformation feature from the objs. This process has a mixed-integer optimization which also takes lots of time and depends on many libraries, such as Comiso, Matlab and gurobi. And do you give me more details about your concerns?

**Comment by uptownfunkmars** [comment]
> yes, it is the same as in the link. It extracts the deformation feature from the objs. This process has a mixed-integer optimization which also takes lots of time and depends on many libraries, such as Comiso, Matlab and gurobi. And do you give me more details about your concerns?

I have installed the dependent environment, including Matlab gurobi. The Comiso is not installed but uses the comiso.dll included in the [link](https://github.com/aldehydecho/convMesh/tree/master/genfeature). Matlab prompts "save file path:. \ 3003 \ featuremat1.mat", but there is no such file in the corresponding path.
